src/BotwinExtensions.cs

```diff
@@ -54,6 +59,34 @@ public static IApplicationBuilder UseBotwin(this IApplicationBuilder builder, Bo
             return builder.UseRouter(routeBuilder.Build());
         }
 
+        private static void Apply405Handler(IApplicationBuilder builder, IEnumerable<BotwinModule> modules)
+        {
+            var systemRoutes = new List<(string verb, string route)>();
+            foreach (var module in modules)
+            {
+                foreach (var route in module.Routes.Keys)
+                {
+                    var strippedPath = route.path.EndsWith("/") ? route.path.Substring(0, route.path.Length - 1) : route.path;
+                    systemRoutes.Add((route.verb, "/" + strippedPath));
+                }
+            }
+
+            builder.Use(async (context, next) =>
+            {
+                var strippedPath = context.Request.Path.Value.EndsWith("/") && context.Request.Path.Value.Length > 1
+                    ? context.Request.Path.Value.Substring(0, context.Request.Path.Value.Length - 1)
+                    : context.Request.Path.Value;
+
+                var verbsForPath = systemRoutes.Where(x => x.route == strippedPath).Select(y => y.verb);
```

Comment:
If you’re looking up routes by paths, then you shouldn’t use a list. Consider an `ILookup<string, string>` instead that maps from paths to the registered verbs.